### ISSUE TITLE
Does this write a new config by design?

### ISSUE THREAD

**Issue by bglidwell** [yeoman/configstore#48]
Every time I run my code, it wipes out the previous config and resets to a blank file with just {}. Looking at the code, this seems to be by design if no defaults are put in. If I put something in default, it writes that instead each time. However, I want my config to persist, such that I can call it up, load some values, change some, close it, and when I rerun my code, it'll pull from that saved config file. It's a bit hard to explain. With my method, I can handwrite the json file if I wanted to. 

This seems to be fixed by commenting out line 28:
	//this.all = assign({}, defaults || {}, this.all || {});

With this line commented out, it loads whatever I have in my config.json file and works great

Just wondering if this is by design, or if we can have an ExistingFile flag to ignore the initialize with {} part

**Comment by sindresorhus** [comment]
It should preserve the existing config. You're either doing something wrong or there's a bug, but I'm leaning towards the former as `configstore` has been working fine for years without problems. Submit a failing tests and I'll take a closer look.

**Comment by bglidwell** [comment]
You are 100% correct. Thank you so much for the fast response. 

My existing .json config file was missing a comma after one of the parameters and once I fixed that, it stopped writing over my config file with {}

**Comment by grischaerbe** [comment]
I think it's not the most graceful method to overwrite a config if the json file is malformed. Read about my use case in the dev-ops software caprover here: [https://github.com/caprover/caprover/issues/858](https://github.com/caprover/caprover/issues/858).
tl;dr: i was manually editing the config and a tiny mistake wiped my config.
Validating json before overwriting it should be the sane way to parse a config file in my opinion.

**Comment by Ynk33** [comment]
We have exactly the same use-case today, and this issue is still here.
I don't understand how wiping the entire file if the json is not valid can be a valid behaviour.
An exception should be raised and the file should not be overwritten at all if its content is not valid json.
